redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts

```diff
@@ -411,6 +416,122 @@ export class RecommendationProvider {
+            let keyType: string;
+            const sortedSetMember = await redisClient.sendCommand(
+              new Command('zrange', [keys[processedKeysNumber].name, 0, 0], { replyEncoding: 'utf8' }),
+            ) as string[];
+            try {
+              keyType = await redisClient.sendCommand(
+                new Command('type', [sortedSetMember[0]], { replyEncoding: 'utf8' }),
+              ) as string;
+            } catch (err) {
+              if (err && checkRedirectionError(err)) {
+                const { address } = parseRedirectionError(err);
+                const nodes = client.nodes('master');
+
+                const node: any = nodes.find(({ options: { host, port } }: Redis) => `${host}:${port}` === address);
+                if (!node) {
+                  throw new ClusterNodeNotFoundError(
+                    ERROR_MESSAGES.CLUSTER_NODE_NOT_FOUND(node),
+                  );
+                }
+
+                keyType = await node.sendCommand(
+                  new Command('type', [sortedSetMember[0]], { replyEncoding: 'utf8' }),
+                ) as string;
+              }
+            }
+            if (keyType === RedisDataType.JSON || keyType === RedisDataType.Hash) {
+              isJSONOrHash = true;
+            }
+            processedKeysNumber += 1;
+            sortedSetNumber += 1;
+          }
+        }
+
+        return isJSONOrHash ? { name: RECOMMENDATION_NAMES.SEARCH_INDEXES } : null;
+      }
+      const sortedSets = keys
+        .filter(({ type }) => type === RedisDataType.ZSet)
+        .slice(0, 100);
+      const res = await redisClient.pipeline(sortedSets.map(({ name }) => ([
+        'zrange',
+        name,
+        0,
+        0,
+      ]))).exec();
+
+      const types = await redisClient.pipeline(res.map(([, member]) => ([
+        'type',
+        member,
+      ]))).exec();
+
+      const isHashOrJSONName = types.some(([, type]) => type === RedisDataType.JSON || type === RedisDataType.Hash);
+      return isHashOrJSONName ? { name: RECOMMENDATION_NAMES.SEARCH_INDEXES } : null;
+    } catch (err) {
+      this.logger.error('Can not determine search indexes recommendation', err);
+      return null;
+    }
+  }
+
+  /*
+ * Check dangerous commands recommendation
+ * @param redisClient
+ */
+
+  async determineDangerousCommandsRecommendation(
+    redisClient: Redis | Cluster,
+  ): Promise<Recommendation> {
+    try {
+      const dangerousCommands = await redisClient.sendCommand(
+        new Command('command', ['LIST', 'FILTERBY', 'aclcat', 'dangerous'], { replyEncoding: 'utf8' }),
+      ) as string[];
+
+      const filteredDangerousCommands = dangerousCommands.filter((command) => {
+        const commandName = command.split('|')[0];
+        return !redisInsightCommands.includes(commandName);
+      });
+
+      const activeDangerousCommands = await Promise.all(
+        filteredDangerousCommands.map(async (command) => await this.checkCommandInfo(redisClient, command)),
```

Comment:
What is the goal to additionally send `command info <cmd>` here?
Is it possible to use `command` without any args and filter all commands from that list?